h2o-py/h2o/model/model_base.py

```diff
@@ -410,6 +410,18 @@ def summary(self):
         print("No model summary for this model")
 
 
+    def model_parameters(self):
```

Comment:
I guess we want to leverage curent `_parms` coming from backend and use usual ways of displaying them. Is this necessary ?

Reply:
@Pscheidl thanks, I will check if parameters could be exposed through _parms

Reply:
It would make your work futureproof.

I had the idea with providing multiple subclasses of GenericModelParameters and their V3 counterparts. This is costly and ineffective.

Just thinking out loud, but having a Map of these values and just creating the V3 objects would be easier (Schema.fillObject would have to be modified for that case).

Or, have one generic V3 serialized. In this case, the code generator would have to be modified as well, but it would be a truly general way and could even be used for my generic metrics system, which is now purely static.

Reply:
@Pscheidl I think that knowing how to convert types of fields within those classes of parameters is one of the most general ways to deserialise. We don't need to care about classes. Right now recursion and reflection are being used and it is not static - ie if the field of supported type is added to a parameters class it will be deserialised just fine. 
I don't get exactly what you described above but I think we can merge current version ( which is done in the same fashion as existing functionality). I'm aware of approaches where json is being deserialised as Map and of course we can create code generators for each class of parameters but don't think it is really necessary. 
I think that there is no `futureproof` solutions as nobody knows future.
Anyway if you want to improve it please go ahead and create proposal PR after this one gets merged. But this proposal might not be something that contributes to extending functionality of the project.